File: src/Pieces/Knight.java
package Pieces;

public class Knight extends Piece {
    public Knight(int y, int x, boolean white) {
        this.x = x;
        this.y = y;
        this.white = white;
    }

    public boolean checkValidMove(int y, int x, Piece[][] board) {
        //Defaulted Moves
        if ((this.x + 2 == x && this.y + 1 == y) || (this.x + 1 == x && this.y + 2 == y) || (this.x - 2 == x && this.y - 1 == y) || (this.x - 1 == x && this.y - 2 == y) || (this.x + 2 == x && this.y - 1 == y) || (this.x + 1 == x && this.y - 2 == y) || (this.x - 2 == x && this.y + 1 == y) || (this.x - 1 == x && this.y + 2 == y)) {
            return true;
        }

        //Defaults False
        return false;
    }
}

File: src/main/kotlin/com/beyondbell/chessAPI/pieces/Knight.kt
package com.beyondbell.chessAPI.pieces

import com.beyondbell.chessAPI.player.Player

class Knight(owner: Player, x: Int, y: Int) : Piece(owner, x, y) {
	override fun checkValidMove(x: Int, y: Int, board: Array<Array<Piece>>): Boolean {
		TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
	}
}